packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -129,6 +130,10 @@ export const useGridKeyboardNavigation = (
         }
 
         case ' ': {
+          const field = (params as GridCellParams).field;
+          if (field === GRID_DETAIL_PANEL_TOGGLE_FIELD) {
```

Comment:
Could probably be a pre-processor like for `canBeReordered` at some point
Should we do the same check for the row grouping / tree data grouping columns ?

Reply:
For tree data it's not a problem, but the check for row grouping is complex (see https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid-pro/src/hooks/features/rowGrouping/useGridRowGrouping.tsx) and I wouldn't duplicate it right now